hermeto/core/models/input.py

```diff
@@ -23,6 +23,37 @@
 ModelT = TypeVar("ModelT", bound=pydantic.BaseModel)
 
 
+def _handle_legacy_allow_binary(
+    instance: Union["PipPackageInput", "BundlerPackageInput"],
+    binary_filter_class: Union[type["PipBinaryFilters"], type["BundlerBinaryFilters"]],
+) -> None:
+    """Handle backward compatibility for allow_binary field."""
+    # If allow_binary is already False, nothing to process
+    if not instance.allow_binary:
+        return
+
+    # Check if user provided both fields originally
+    user_provided_both = instance.allow_binary and instance.binary is not None
+    # Determine if allow_binary should be migrated to binary field
+    should_migrate_allow_binary = instance.allow_binary and instance.binary is None
+
+    if user_provided_both:
+        log.warning(
+            "Both 'allow_binary' and 'binary' fields specified. "
+            "The 'binary' field will take precedence. "
+            "Please remove 'allow_binary' as it is deprecated."
+        )
+    elif should_migrate_allow_binary:
+        log.warning(
+            "The 'allow_binary' field is deprecated and will be removed in the next major version. "
+            "Please use 'binary': {} instead of 'allow_binary': true."
```

Comment:
Should it be something like `'binary': {'arch...}` or do you mean that an empty map is fine too?